packages/ui/textEditor/Lexical/plugins/LinkEditor/useModal.jsx

```diff
@@ -0,0 +1,51 @@
+import { useCallback, useMemo, useState } from "react";
+import * as React from "react";
+import { Overlay } from "@figshare/fcl/overlay";
+
+import styles from "./useModal.css";
+
+
+export default function useModal() {
+  const [modalContent, setModalContent] = useState(null);
+
+  const onClose = useCallback(() => {
+    setModalContent(null);
+  }, []);
+
+  const modal = useMemo(() => {
+    if (modalContent === null) {
+      return null;
+    }
+
+    const { title, content, isVisible } = modalContent;
+
+    return (
+      <Overlay
+        classNameContainer={styles.editorOverlayContainer}
+        isVisible={isVisible}
+        portalNode='div[data-id="editor-content-editable"]'
+        title={title}
+        onClose={onClose}
+      >
+        {content}
+      </Overlay>
+    );
+  }, [modalContent, onClose]);
+
+  const showModal = useCallback(
+    (
+      title,
```

Comment:
@alinacoca Interesting approach. using a `hook` to mount a component and it's content 🤔 

I would suggest the argument for this function be an `object`:
`useCallback({ title, getContent, isVisible }) => { ... }, [onClose, setModalContent])` - not sure what `isVisible` is used for. will there be instances where isVisible is false ?